File: src/main/java/com/PrzeBarCore/Laboratorymanagementsystem/dto/events/PublishedEvent.java
package com.PrzeBarCore.Laboratorymanagementsystem.dto.events;

import com.PrzeBarCore.Laboratorymanagementsystem.global.AggregateType;
import com.PrzeBarCore.Laboratorymanagementsystem.global.EventType;
import tools.jackson.databind.JsonNode;

import java.time.LocalDateTime;

public record PublishedEvent(Long eventId, EventType eventType, AggregateType aggregateType, Long aggregateId, LocalDateTime createdAt, JsonNode payload) {
}

File: order-monitoring-worker/src/main/kotlin/com/PrzeBarCore/ordermonitoringworker/dto/PublishedEvent.kt
package com.PrzeBarCore.ordermonitoringworker.dto

import com.PrzeBarCore.ordermonitoringworker.global.enums.AggregateType
import com.PrzeBarCore.ordermonitoringworker.global.enums.EventType
import tools.jackson.databind.JsonNode
import java.time.LocalDateTime

data class PublishedEvent (val eventId: Long, val eventType: EventType, val aggregateType: AggregateType, val aggregateId : Long, val createdAt: LocalDateTime, val payload : JsonNode) {
}